src/main/java/com/example/Restaurant_Management_System_REST_API/model/entity/Customer.java

```diff
@@ -0,0 +1,73 @@
+package com.example.Restaurant_Management_System_REST_API.model.entity;
+
+
+import jakarta.persistence.*;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+import org.springframework.security.core.authority.SimpleGrantedAuthority;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.userdetails.UserDetails;
+
+import java.util.Collection;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+
+@Entity
+@Table(name = "application_users")
+@Getter
+@Setter
+@AllArgsConstructor
+@NoArgsConstructor
+public class Customer implements UserDetails {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    @Column(name = "id", nullable = false)
+    private Long id;
+
+    //Below are fields for security issues
+    private String password;
+    private String emailAddress; //TODO: delete when abstract class with contact details will be implemented
+    @ElementCollection(fetch = FetchType.EAGER)
+    private Set<String> roles;
+
+    @Override
+    public Collection<? extends GrantedAuthority> getAuthorities() {
+        return roles.stream()
+                .map(SimpleGrantedAuthority::new)
+                .collect(Collectors.toList());
+    }
```

Comment:
This part of code will return new role every time this method is used. I think that's not how it should be implemented.

Reply:
you are right! I think I should move this creation of a new Role to the constructor of Customer. Then each time an instance of Customer is created a new role will be given, right?

Reply:
nope this was a bad idea, I made it differently